content/guides/virtualizing/bhyve.md

```diff
@@ -0,0 +1,158 @@
+
+Before proceeding with the tutorial, you will have to make a few system configurations in order to be able to move on in a smooth manner.
+
+##### CA Certificates
+
+Since Haiku images can be fetched securely only via [HTTPS](https://en.wikipedia.org/wiki/HTTPS), you will need to download some [CA root certificates](https://en.wikipedia.org/wiki/Certificate_authority), as FreeBSD does not ship with them by default. In order to do that, you will need to install the `security/ca_root_nss` package, which is a part of the [FreeBSD ports](https://www.freebsd.org/ports/).
+
+```sh
+# feel free to skip this process if you have done it before.
+portsnap fetch
+portsnap extract
+
+cd /usr/ports/security/ca_root_nss
+make install clean
+```
+
+Now, follow through the process, and you will be able to fetch the image file securely.
+
+##### Networking & Kernel Preparations <a name="part_download"></a>
+
+Since `bhyve` consists of kernel modules that ship with FreeBSD by default, you do not need to install the core part of `bhyve`.
+
+However, you will need to load the modules manually in order to use `bhyve`.
+
+```sh
+kldload vmm
+kldload nmdm
+kldload if_bridge
+kldload if_tap
+```
+
+However, the kernel modules will only remain loaded until you reboot your machine. In order to make the changes permanent, edit `/boot/loader.conf` by adding:
+
+```sh
+# Kernel modules
+vmm_load="YES"
+nmdm_load="YES"
+bridge_load="YES"
+tap_load="YES"
+```
+
+Now, run the following commands in order to create a network device and a bridge for your virtual machine, in order to make a network bridge (`bridge0`) that is linked to the virtual device `tap0`.
+
+```sh
+sysctl net.link.tap.up_on_open=1
+sysctl net.inet.ip.forwarding=1
+
+ifconfig tap0 create
+ifconfig bridge0 create
+
+ifconfig bridge0 addm tap0
+```
+
+But the problem is that the device `tap0` itself is not linked to any physical, working internet source. This means that even though you may have made the network bridge, you still need to link `tap0` to a physical network device. In my case, the device that I need to use is `re0`, but you can find yours by running `ifconfig`.
+
+```sh
+ifconfig bridge0 addm re0
+```
+
+Lastly, you want to run the following:
+
+```sh
+ifconfig bridge0 up
+```
+
+Now, make your changes permanent by editing `/etc/sysctl.conf/` and appending the following:
+
+```sh
+sysctl net.link.tap.up_on_open=1
+sysctl net.inet.ip.forwarding=1
+```
+
+
+### Downloading the Haiku Anyboot image <a name="part_download">
+
+Now that you have made the necessary changes to your system, you can now obtain a Haiku anyboot image securely, which will be necessary for the creation of a Haiku virtual machine. You will need to install `wget`, in order to fetch the image file from the internet. Because of the fact that Haiku image files are packed alongside with a `ReadMe` file inside of a `.zip` archive, you will need the `unzip` package in order to extract the image file.
```

Comment:
```suggestion
Now that you have made the necessary changes to your system, you can securely obtain a Haiku anyboot image. You will need to install `wget` to download the image file and because it's packed in a  `.zip` archive together with a `ReadMe` file, you will need the `unzip` package to extract it.
```